lua/apisix/plugins/proxy-rewrite.lua

```diff
@@ -110,6 +115,18 @@ function _M.rewrite(conf, ctx)
     end
 end
 
+function _M.header_filter(conf, ctx)
+    if ngx.status >= 300 then
+        return
+    end
+
+    if conf.resp_headers then
+        for header_name, header_value in pairs(conf.resp_headers) do
```

Comment:
avoid to use `pairs`,  it a NYI: http://wiki.luajit.org/NYI

Reply:
also this line has the same issue, 
```
for header_name, header_value in pairs(conf.headers) do
```

how about we change the scheme of  headers and  resp_headers into array like 
```
{
  {
    field = 'X-Server',
    value = 'Hello'
  }
}
```

Reply:
@membphis

Reply:
Yes, that is better.